Refactors Lich Phylactery to be a component, makes it actually work

MrMelbert opened the pull request:
## About The Pull Request

- Refactors the Lich's Phylactery mechanic to be a component
  - The phylactery now works on signals instead of processing and waiting for their owner to die. 
- Fixes Lich Phylactery reviving not working
- Minor code improvements for things surrounding lich code
  - Made the IMBUE_SOUL signal return a bitflag instead of a boolean
- Also overall improves the UX for the phylactery. 
  - Better examine messages
  - Feedback for the lich when they die
- Updates the description for the bind soul spell to better reflect it's behavior. Also fixes said behavior. 
  - Apparently at some point someone completely reworked how Lich works but didn't update the spell description at all (FIVE YEARS AGO), leaving it to be completely misleading. I chose to restore part of the behavior and update the description to match how it should... probably work? I'm not entirely sure

## Why It's Good For The Game

Lich's phylactery straight up like, didn't work
This makes it work, and improves the code a good bit. 

## Changelog

:cl: Melbert
fix: Makes the wizard's Lich Phylactery mechanic actually work consistently. 
fix: The description of the Bind Soul spell (and entry) should match its behavior a bit more accurately now. You will revive regardless if your body is destroyed or leaves the z-level. With each revival, the time it takes to respawn does not change, but the duration of the stun does. 
refactor: Makes Phylacteries a component.
qol: Gave Lich Phylacerties some QoL features and pizazz. Examining it'll offer more information to wizards and ghosts.
/:cl:

MrMelbert committed:
Refactors phylactery to be a component
- Overall improves lich code
- Changed the SOUL_IMBUE signal to return a bitflag
- Minor style  improvements (indentation and such)

MrMelbert committed:
Tweak initialize + fixes a bug with being gibbed

MrMelbert committed:
adds a clarification comment

MrMelbert committed:
And fixes a compile error oops

MrMelbert committed:
reorganizing

MrMelbert committed:
clear the revive timer, just in case

MrMelbert committed:
oops this should drop stuff

MrMelbert committed:
missing no soul trait

MrMelbert committed:
type in a variable name

MrMelbert committed:
god damn devil removal
- this check was removed when devil was and i think it's kinda important to prevent stacking lichdom

MrMelbert committed:
rework this a bit

MrMelbert committed:
plurals

MrMelbert committed:
also, wrong source

MrMelbert committed:
updates the description

SuperNovaa41 commented on code/datums/components/phylactery.dm:
```diff
@@ -0,0 +1,216 @@
+	/// How much time is added on to the respawn time per revival.
+	var/time_per_resurrection = 1 MINUTES
+	/// The color of the phylactery itself. Applied on creation.
+	var/phylactery_color = "#003300"
```
```suggestion
	var/phylactery_color = COLOR_VERY_DARK_LIME_GREEN
```

SuperNovaa41 commented on code/datums/components/phylactery.dm:
```diff
@@ -0,0 +1,216 @@
+/**
+ * Signal proc for [COMSIG_PARENT_EXAMINE].
+ *
+ * Gives some flavor fort he phylactery on examine.
```
```suggestion
 * Gives some flavor for the phylactery on examine.
```
wow

SuperNovaa41 commented on code/datums/components/phylactery.dm:
```diff
@@ -0,0 +1,216 @@
+	UnregisterSignal(source, COMSIG_LIVING_REVIVE)
+
+/**
+ * Actuall undergo the process of reviving the lynch
```
The what now?

MrMelbert committed:
Some review + undoes some changes
- Corrects some comments
- Uses defines
- Rewords the lichdom entry and spell desc
- Sets default time per res to 0 secs to avoid changing behaviors

timothymtorres commented:
@MrMelbert did this solve the issues for #62507, #37808, or #45518?

MrMelbert commented:
fixed A, not sure about B, did not fix C
